### ISSUE TITLE
Calling IEnumerable.count() twice gets different results

### ISSUE THREAD

**Issue by XC-Zhang** [mkloubert/node-enumerable#1]
Here is the code I use to reproduce the issue.

```javascript
const Enumerable = require("node-enumerable");
const assert = require("assert");
const data = Enumerable.from([1, 2, 3, 4, 5]);
assert.equal(data.count(), data.count());
```

The code above throws an assertion error.

```
AssertionError: 5 == 0
```

Environment
- node v6.11.2
- node-enumerable v3.6.0

Is it expected?
Thanks in advance! 😃

**Comment by mkloubert** [comment]
@XC-Zhang

Yes. You have to call reset() before you can invoke count() with the same result, because the pointer is at the end after the first call.

So it always works the same way for strings, Arrays and generators.

Keep in mind: reset() does not work with generators, because JavaScript does not support going back to the beginning.

You can check canReset property of a sequence to check if it is possible to call reset().

If not: You can invoke makeResettable() which converts a sequence to an array based sequence, which can make use of reset().

**Comment by XC-Zhang** [comment]
Thanks for quick response and detailed explanation! I didn't notice the `reset()` method. Now it solves the problem. 
Thanks again! 😄

**Comment by mkloubert** [comment]
@XC-Zhang 

Btw.: I released a [new version 3.7.1](https://github.com/mkloubert/node-enumerable/blob/master/CHANGELOG.md#371-november-7th-2017-methods), with some new methods like the [length() method](https://github.com/mkloubert/node-enumerable#count-).

It will always return the same value (without calling `reset()`), if you have a sequence based on an array-like object, like a string or an array:

```javascript
const Enumerable = require("node-enumerable");
const assert = require("assert");
const data = Enumerable.from([1, 2, 3, 4, 5]);

assert.equal(data.length(),
             data.length());
```

If it is something else, like a generator, it will behave as `count()`.
